lib/Test/PostgreSQL.pm

```diff
@@ -116,7 +116,42 @@ has socket_dir => (
   is => "ro",
   isa => Str,
   lazy => 1,
-  default => method () { File::Spec->catdir( $self->base_dir, 'tmp' ) },
+  default => method () {
+    my $dir = File::Spec->catdir( $self->base_dir, 'tmp' );
+
+    # This magic number is based on the Unixy systems limit for
+    # Unix socket path lengths: on Linuxes it is at 108 characters,
+    # MacOS allows no more than 103. Considering that we are
+    # generating a directory path, and the actual socket file name
+    # will be appended to it, 80 characters seemed a safe assumption.
+    if (length $dir > 80) {
+      $self->{_socket_dir} = File::Temp->newdir(
+        'pgtest_sock.XXXXX',
+        
+        # base_dir is preserved if environment variable
+        # TEST_POSTGRESQL_PRESERVE is truthy; this is done
+        # for diagnostics. Socket directory should always
+        # be cleaned up since there is nothing of interest there.
+        CLEANUP => 1,
+        TMPDIR  => 1,
+
+        # TODO Come up with something better if this ever cause
+        # any trouble. Straightforward approach with using File::Temp
+        # that consults $TMPDIR environment variable is exactly
+        # the cause for the Unix socket path being too long;
+        # other approaches at sniffing what system we're running on
+        # and what temporary paths are available also fail at various
+        # edge cases, which is exactly the reason File::Temp was created
+        # in the first place. So, no easy answer here; simply falling back
+        # to the timeless default.
+        DIR => '/tmp',
```

Comment:
I made this change assuming that we only support Unixy systems; only after pushing the branch I realized that we do try to support Windows. @SysPete Are we actually using Unix sockets on Windows? It's been a long time since I've done any development on that platform so I'm not sure if Unix sockets are supported there. If they are not then `socket_path` is likely irrelevant and this hardcoded value shouldn't cause any trouble; otherwise I'll need to come up with something more compatible because breaking existing Windows support is not my goal.